### ISSUE TITLE
Pagination fails: wrong type for value; expected string; got template.HTML

### ISSUE THREAD

**Issue by MarkEdmondson1234** [digitalcraftsman/hugo-icarus-theme#70]
Hi, lovely theme.

Its all looking good save when I have enough posts to trigger pagination, in which case I get this error when building the site:

```
Started building site
ERROR: 2017/01/15 06:43:32 template.go:131: template: theme/partials/pagination.html:9:73: executing "theme/partials/pagination.html" at <absURL>: wrong type for value; expected string; got template.HTML in theme/partials/pagination.html
ERROR: 2017/01/15 06:43:32 template.go:131: template: theme/partials/pagination.html:4:73: executing "theme/partials/pagination.html" at <absURL>: wrong type for value; expected string; got template.HTML in theme/partials/pagination.html
ERROR: 2017/01/15 06:43:32 template.go:131: template: theme/partials/pagination.html:9:73: executing "theme/partials/pagination.html" at <absURL>: wrong type for value; expected string; got template.HTML in theme/partials/pagination.html
ERROR: 2017/01/15 06:43:32 template.go:131: template: theme/partials/pagination.html:4:73: executing "theme/partials/pagination.html" at <absURL>: wrong type for value; expected string; got template.HTML in theme/partials/pagination.html
```

Looking in the template it seems the `.Paginator.Prev.URL` and `.Paginator.NextURL` are not found:

```
<nav id="page-nav">
{{ if or (.Paginator.HasPrev) (.Paginator.HasNext) }}
    {{ if .Paginator.HasPrev }}
        <a class="extend prev" rel="prev" href="{{ .Paginator.Prev.URL | absURL }}">
            « {{ with .Site.Data.l10n.pagination.previous }}{{ . }}{{ end }}
        </a>
    {{ end }}
    {{ if .Paginator.HasNext }}
        <a class="extend next" rel="next" href="{{ .Paginator.Next.URL | absURL }}">
            {{ with .Site.Data.l10n.pagination.next }}{{ . }}{{ end }} »
        </a>
    {{ end }}
{{ end }}
</nav>
```

which produces the following HTML breaking layout:

```
nav id="page-nav">

    
    
        <a class="extend next" rel="next" href="
</section>
```

I'm new to Hugo so think I could be missing something obvious, but any help appreciated :)

**Comment by digitalcraftsman** [comment]
Hello @MarkEdmondson1234,

this is unusual. I tested it with Hugo 0.18.1 without problems and the if statements should check (twice) if the links actually exist.

- did you customized the theme somehow?
- which version of Hugo are you running? Just post the output from `hugo env`.

The easiest way to reproduce this error would be to share your setup if this is possible.

**Comment by MarkEdmondson1234** [comment]
Thanks @digitalcraftsman - it is customised as in its done via R in https://github.com/rstudio/blogdown, but I don't see how that would affect just pagination, but perhaps.  I have the website here: https://github.com/MarkEdmondson1234/markedmondson.me-hugo

It installed hugo via its `blogdown::install_hugo()` command, when I first try `hugo env` I get an error (uh oh?) but `hugo version`gives this:

```
foo:~ mark$ hugo version
Hugo Static Site Generator v0.16 BuildDate: 2017-01-14T22:18:39+01:00
foo:~ mark$ hugo env
Error: unknown command "env" for "hugo"
```

**Comment by digitalcraftsman** [comment]
Thanks. `hugo env` was implemented in one of the newer releases (>0.16). Nonetheless, I cloned the repo with the source files of your blog, changed the pagination to `paginate = 1` in the config file and built your site with Hugo v0.18.1. 

With this setup I faced no issues with the pagination. Maybe you can confirm this by upgrading to the latest version of Hugo.

**Comment by MarkEdmondson1234** [comment]
Great thanks, that was it.

I used the [brew OSX install](https://gohugo.io/overview/installing/) so:

```
brew unlink hugo
brew update && brew install hugo
hugo env

#> Hugo Static Site Generator v0.18.1 BuildDate: 2016-12-29T18:12:41+01:00
#> GOOS="darwin"
#> GOARCH="amd64"
#> GOVERSION="go1.7.4"
```
